OverlayPlugin.Core/MemoryProcessors/PartyList/PartyListMemory62.cs

```diff
@@ -0,0 +1,291 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Runtime.CompilerServices;
+using System.Runtime.InteropServices;
+using Newtonsoft.Json.Linq;
+using RainbowMage.OverlayPlugin.MemoryProcessors.AtkGui.Managed;
+using RainbowMage.OverlayPlugin.MemoryProcessors.Combatant;
+
+namespace RainbowMage.OverlayPlugin.MemoryProcessors.PartyList
+{
+    interface IPartyListMemory62 : IPartyListMemory { }
+
+    class PartyListMemory62 : PartyListMemory, IPartyListMemory62
+    {
+        public PartyListMemory62(TinyIoCContainer container) : base(container) { }
+
+        public override Version GetVersion()
+        {
+            return new Version(6, 2);
+        }
+
+        public unsafe AtkGui.Managed.AddonPartyList GetPartyList()
+        {
+            var blankList = new AtkGui.Managed.AddonPartyList();
+            if (!IsValid())
+            {
+                return blankList;
+            }
+
+            if (!atkStageMemory.IsValid())
+            {
+                return blankList;
+            }
+
+            var partyListPtr = atkStageMemory.GetAddonAddress("_PartyList");
+
+            if (partyListPtr != IntPtr.Zero)
+            {
+                return ReadMemory_AddonPartyList(partyListPtr);
+            }
+
+            return blankList;
+        }
+
+        private unsafe AtkGui.Managed.AddonPartyList ReadMemory_AddonPartyList(IntPtr ptr)
+        {
+            if (ptr.ToInt64() == 0)
+            {
+                return null;
+            }
+            var managed = new AtkGui.Managed.AddonPartyList();
+            managed.Pointer = ptr.ToInt64();
+
+            byte[] source = memory.GetByteArray(ptr, Marshal.SizeOf<AtkGui.Unmanaged.AddonPartyList>());
+
+            fixed (byte* p = source)
+            {
+                var memory = *(AtkGui.Unmanaged.AddonPartyList*)&p[0];
+
+                ConvertMemory_AddonPartyList(managed, memory);
+            }
+
+            return managed;
+        }
+
+        private unsafe AtkGui.Managed.AddonPartyList ConvertMemory_AddonPartyList(AtkGui.Managed.AddonPartyList managed, AtkGui.Unmanaged.AddonPartyList memory)
```

Comment:
It'd sure be nice if this could be more automatic.

Reply:
I've got a much better solution for this that handles everything via DynamicObject, I'm just trying to work out the proper way to get a fixed byte array back from the compile-time boxing process that results in the type of the array going from e.g. `fixed byte Name[0x20]` to `AtkUnitBase+<Name>e__FixedBuffer`.

Reply:
See `ManagedType` in force pushed commit. As well, see `AtkStageMemory62` and `PartyListMemory62` for usage examples.

It's important to note that running `ToJObject` on the entire `AddonPartyList` object takes about 8-9 seconds due to a lot of very small, discrete reads of the game memory. As such, usage of that method should be avoided unless explicitly needed, e.g. the `getPartyList` event handler should traverse the object C#-side to only fetch the required information to get party member info rather than passing the entire `AddonPartyList` object back to the calling overlay.

Reply:
I see, so ManagedType lazily reads memory when you access fields. Neat!

This is a thought for later, but is it the individual reads that have higher overhead, in the sense that you could do an optimization where you intentionally do a larger read once and cache it here in case there's sequential reads in memory in a common case?

Reply:
The slowdown is caused by making a lot of small reads, rather than one large read. However, all of the memory read here is dynamically allocated, so it's not really feasible to pull in large chunks unless those large chunks amount to several hundred megabytes.